src/libData/BlockChainData/BlockChain.h

```diff
@@ -104,6 +104,18 @@ class BlockChain {
 
     if (blockNumOfExistingBlock < blockNumOfNewBlock ||
         INIT_BLOCK_NUMBER == blockNumOfExistingBlock) {
+      if (m_blocks.size() != 0) {
+        uint64_t blockNumOfLastBlock =
+            m_blocks[blockNumOfNewBlock - 1].GetHeader().GetBlockNum();
```

Comment:
what if blockNumOfNewBlock -1 does not exist ? it will use dummy block (uint64_t - 1) ?